lib/src/services/moyasar_service.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:moyasar/moyasar.dart';
+import 'package:moyasar/src/models/payment_request.dart';
+import 'package:moyasar/src/models/sources/apple_pay/apple_pay_request_source.dart';
+import 'package:moyasar/src/models/sources/card/card_request_source.dart';
+import 'package:moyasar/src/moyasar.dart';
+
+class MoyasarService {
+  static Future<PaymentResponse> pay({
+    required PaymentConfig config,
+    required CardFormModel cardData,
+  }) async {
+    try {
+      final CardPaymentRequestSource source =
+          CardPaymentRequestSource(cardData);
+      final PaymentRequest paymentRequest = PaymentRequest(
+        config,
+        source,
+      );
+      final PaymentResponse result = await Moyasar.pay(
+        apiKey: config.publishableApiKey,
+        paymentRequest: paymentRequest,
+      );
+      return result;
+    } catch (e) {
+      rethrow;
```

Comment:
On line 46 we use `return PaymentStatus.failed;` but here we use `rethrow`.
Why are different approaches used?
Maybe we should do the same thing in these two places?